test_provider.py

```diff
@@ -0,0 +1,149 @@
+        atom.observed = json['metadata:complete']
+        atom.qa_state = ...
+        atom.guide_state = ...
+        atom.required_resources = ...
+        atom.wavelength = ...
+        return atom
+
+    @staticmethod
+    def parse_sidereal_target(json: dict) -> SiderealTarget:
+        sidereal = SiderealTarget()
+        sidereal.ra = json['ra'] # this might need transformation
+        sidereal.dec = json['dec'] # this might need transformation
+        sidereal.pm_ra = json['base']['deltara']
+        sidereal.pm_dec = json['base']['deltadec']
+        sidereal.epoch = json['base']['epoch']
+        return sidereal
+
+    @staticmethod
+    def parse_nonsidereal_target(json: dict) -> NonsiderealTarget:
+        nonsidereal = NonsiderealTarget()
+        nonsidereal.des = json['base']['des']
+        nonsidereal.tag = json['base']['tag']
+        nonsidereal.ra = ...
+        nonsidereal.dec = ...
+
+        return nonsidereal
+
+    @staticmethod
+    def parse_observation(json: dict) -> Observation:
+        obs = Observation()
+
+        obs.id = json['observationId']
+        obs.internal_id = json['key']
+        obs.title = json['title']
+        obs.site = ...
+        obs.status = json['obsStatus']
+        obs.active = ...
+        obs.priority = json['priority']
+        obs.setuptime_type = SetupTimeType(json['setupTimeType'])
+        obs.acq_overhead = ...
+        obs.obs_class = ObservationClass(json['obsClass'])
+        obs.exec_time = ...
+        obs.program_used = ...
+        obs.partner_used = ...
+        targets_in_json = [json[key] for key in json.keys() if key.startswith('TELESCOPE_TARGETENV')]
+        targets = [JsonProvider.parse_sidereal_target(target) if target['base']['tag'] == 'sidereal' else JsonProvider.parse_sidereal_target(target) for target in targets_in_json]
+        obs.targets = targets
+        obs.guide_stars = ...
+        obs.sequence = [JsonProvider.parse_atom(seq) for seq in json['sequence']]
+        find_constraints = [json[key] for key in json.keys() if key.startswith('SCHEDULING_CONDITIONSt')]
+        obs.constraints = JsonProvider.parse_constraints(find_constraints[0])
+        obs.too_type = ...
+    
+    @staticmethod
+    def parse_time_allocation(json: dict) -> TimeAllocation:
+        print(json['timeAccountAllocationCategories'])
+        ta = TimeAllocation(TimeAccountingCode(json['timeAccountAllocationCategories'][0]['category']),
+                            timedelta(milliseconds=json['awardedTime']),
+                            timedelta(milliseconds=0),
+                            timedelta(milliseconds=json['timeAccountAllocationCategories'][0]['programTime']),
+                            timedelta(milliseconds=json['timeAccountAllocationCategories'][0]['partnerTime']))
+        return ta
+
+
+    @staticmethod
+    def parse_or_group(json: dict) -> OrGroup:
+        ...
+
+    @staticmethod
+    def parse_and_group(json: dict) -> AndGroup:
+        ...
```

Comment:
These should be pretty easy for the ODB, I think. We don't need to implement `parse_or_group` and `parse_and_group` will just correspond to scheduling groups and possibly organizational folders.

We will need special logic in `Observation` and `Program` to make sure that the root group and the terminal node groups are handled properly, but I don't think this should be too hard.

Let me know if you're having problems with this and we can discuss.